grype/presenter/sarif/presenter_test.go

```diff
@@ -132,6 +131,103 @@ func createDirPresenter(t *testing.T) *Presenter {
 	return pres
 }
 
+func Test_locations(t *testing.T) {
```

Comment:
I think this test is okay for now. ✅  If you're looking to make tests more idiomatic to Go in the future, here are a few thoughts for your consideration:

- It looks like this is testing two independent receiver methods of the `sarif.Presenter` type — `Presenter.inputPath()` and `Presenter.packagePath`. This ends up being a nice way to organize the tests, where the test name can reflect the receiver method name. So instead of `Test_locations`, you might have `Test_Presenter_inputPath` and `Test_Presenter_packagePath`.
- Table-driven tests tend to be a good fit when you have repeating patterns in test setups and assertions. So typically you'd start by defining a slice of test cases, which each test case including a name for the test case, the input data, and then the expected data. And then after that slice is defined, you just just need to define the setup and assertion implementation once, because that can be done inside a `for` loop that iterates over the test cases.

Reply:
Right, these tests just seemed to be every-so-slightly different, which is why I didn't go with the table-driven tests, and these things are quite interrelated which is why I did not separate them into different methods, since I'd end up testing the latter (packgePath) quite a lot with different inputPath permuations.